### PR TITLE
lldb debugger support in realgud package

### PR DESCRIPTION
**Pull request by rocky**
### Brief summary of what the package does

Adds support for the lldb debugger inside the realgud debugger front-end package

[Please write a quick summary of the package.]

### Direct link to the package repository

https://github.com/realgud/realgud-lldb

### Your association with the package

I am the author and maintainer. 

### Relevant communications with the upstream package maintainer

[e.g., `package.el` compatibility changes that you have submitted. Write **None needed** if there is no problem.]

**None needed**

There are at least two more realgud packages in the javascript space, but it is best to learn the ropes and get feedback with this and realgud-ipdb before starting the others.

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

### CODE REVIEW

File: recipes/realgud-lldb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+(realgud-lldb
+ :fetcher github
+ :repo "realgud/realgud-lldb"
+ :files ("realgud-lldb.el"
```
Comment (purcell) [inline recipes/realgud-lldb]: I'll just edit this to use `:defaults` instead of this specific filename, and then merge.

**Comment by purcell** [comment]
I merged this a bit prematurely because the `.el` file is lacking a package dependency on `load-relative`. Can you add that please?

**Comment by rocky** [comment]
Ooops - it should be there now in https://github.com/realgud/realgud-lldb/commit/2296cc465f98d8fdd3b7af2b0dc167feb899b1be  

I also added a dependency on cl-lib. If that shouldn't be there, let me know and I will remove it. Do  I also need a dependency on test-simple for testing? (I haven't converted to ert yet). 

Please double check that I did this right. I make a lot of inadvertent mistakes - that is one reason I work on debuggers. 
 
Sorry for all of the trouble and thanks for all of your help.

**Comment by purcell** [comment]
> Do I also need a dependency on test-simple for testing? (I haven't converted to ert yet)

No, because we don't bundle `*-test.el` or `*-tests.el` files.

> Please double check that I did this right. I make a lot of inadvertent mistakes - that is one reason I work on debuggers.

It all looks good.

> Sorry for all of the trouble and thanks for all of your help.

No trouble at all.

**Pull request merged**